core/dbt/adapters/base/relation.py

```diff
@@ -194,6 +195,15 @@ def render(self) -> str:
         # if there is nothing set, this will return the empty string.
         return ".".join(part for _, part in self._render_iterator() if part is not None)
 
+    def render_limited(self) -> str:
+        rendered = self.render()
+        if self.limit is None:
```

Comment:
Do we need this if we're also doing this check in `__str__`?

Reply:
technically it's fine to remove, but this keeps the methods independent from one another and `render_limited` could be called from other places as well. If we remove this & make it less defensive / more context-aware, i'd just say we make it a 'private' method `_render_limited`